core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -147,8 +150,17 @@
 @Getter
 public class GeyserSession implements GeyserConnection, CommandSender {
 
+    /**
+     * Id used in protocol to refer to primary client (0) or splitscreen subclients (>0)
+     */
+    private final int clientId;
+
+    @Setter
+    private ECPublicKey identityPublicKey;
+
     private final @Nonnull GeyserImpl geyser;
     private final @Nonnull UpstreamSession upstream;
+    private final @Nonnull EventLoopGroup eventLoopGroup;
```

Comment:
While I was testing it I found that having all the spit screen sessions on the same thread made it very laggy, for example blocks would take a second or two to register as being broken, mobs would be delayed in taking damage etc. I'm not sure if this was because of being on the same thread, but this approach did seem to improve performance. So I'm not sure what a better alternative would be?

Reply:
Have changed it back to use the same eventLoop for all subclient sessions